browser-test/src/support/admin_questions.ts

```diff
@@ -14,6 +14,13 @@ export class AdminQuestions {
     await this.expectAdminQuestionsPage();
   }
 
+  async clickSubmitButtonAndNavigate(buttonText: string) {
+    await Promise.all([
+      this.page.waitForNavigation(),

```

Comment:
Curiosity question...

I see that this.page.waitForNavigation is called above the button click. My understanding is that we want to wait for navigation **after** the button click. How do we know that the waitForNavigation will happen after the button click?  I assume it's something to do with this being an async function and the Promise type?

Reply:
Good question. This seems to be a fairly common design pattern e.g., https://playwright.dev/docs/events/ where you asynchronously start listening for the result, then asynchronously trigger the event, and then use await Promise.all to wait for both async events to complete. The reason the waitForNavigation comes first is that you don't want there to be even a tiny gap between where you kick off the event and before you listen for the result. Listening first ensures you'll catch the result and since it's async it does a non-blocking wait while you trigger the click event.

Perhaps a better explanation here: https://github.com/microsoft/playwright/issues/5470

What's confusing is that Playwright functions do a lot of magical waiting themselves so it can be a bit hard to tell where this is essential to do.